### PR TITLE
add login api

### PR DESCRIPTION
**Pull request by corgiboygsj**

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/structure/auth/LoginResult.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+package com.baidu.hugegraph.structure.auth;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class LoginResult {
+
+    @JsonProperty("token")
+    private String token;
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/auth/LoginResult.java:27]: add user_id?

Comment (corgiboygsj) [reply src/main/java/com/baidu/hugegraph/structure/auth/LoginResult.java:27]: only return token in login method

File: src/main/java/com/baidu/hugegraph/api/auth/LoginAPI.java [hunk 2]
```diff
@@ -45,11 +45,13 @@ public LoginResult login(Login login) {
     }
 
     public void logout() {
-        this.client.delete(this.path(), Maps.newHashMap());
+        String path = String.join("/", this.path(), "logout");
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/auth/LoginAPI.java:48]: also update login path

File: pom.xml [hunk 3]
```diff
@@ -53,7 +53,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <compiler.source>1.8</compiler.source>
         <compiler.target>1.8</compiler.target>
-        <hugegraph.common.version>1.8.4</hugegraph.common.version>
+        <hugegraph.common.version>1.8.9</hugegraph.common.version>
```
Comment (Linary) [inline pom.xml:56]: upgrade client version to 1.9.5

**Review by Linary — CHANGES_REQUESTED** [review]

File: src/main/java/com/baidu/hugegraph/api/auth/TokenVerifyAPI.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    public TokenVerifyAPI(RestClient client, String graph) {
+        super(client, graph);
+    }
+
+    @Override
+    protected String type() {
+        return HugeType.TOKEN_VERIFY.string();
+    }
+
+    public Map<String, Object> verifyToken() {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/auth/TokenVerifyAPI.java:39]: define a class and return its instance

File: src/main/java/com/baidu/hugegraph/api/auth/TokenVerifyAPI.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+package com.baidu.hugegraph.api.auth;
+
+import java.util.Map;
+
+import com.baidu.hugegraph.client.RestClient;
+import com.baidu.hugegraph.rest.RestResult;
+import com.baidu.hugegraph.structure.constant.HugeType;
+
+public class TokenVerifyAPI extends AuthAPI {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/auth/TokenVerifyAPI.java:28]: prefer TokenAPI

File: src/main/java/com/baidu/hugegraph/driver/AuthManager.java [hunk 6]
```diff
@@ -54,6 +58,8 @@ public AuthManager(RestClient client, String graph) {
         this.accessAPI = new AccessAPI(client, graph);
         this.belongAPI = new BelongAPI(client, graph);
         this.loginAPI = new LoginAPI(client, graph);
+        this.logoutAPI = new LogoutAPI(client, graph);
+        this.tokenVerifyAPI = new TokenVerifyAPI(client, graph);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/driver/AuthManager.java:62]: rename to tokenAPI

File: src/main/java/com/baidu/hugegraph/api/auth/TokenAPI.java [hunk 7]
```diff
@@ -36,8 +37,8 @@ protected String type() {
         return HugeType.TOKEN_VERIFY.string();
     }
 
-    public Map<String, Object> verifyToken() {
+    public TokenPayload verifyToken() {
         RestResult result = this.client.get(this.path());
-        return result.readObject(Map.class);
+        return TokenPayload.fromMap(result.readObject(Map.class));
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/auth/TokenAPI.java:42]: result.readObject(TokenPayload.class)

File: src/main/java/com/baidu/hugegraph/structure/auth/TokenPayload.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    public String userId() {
+        return this.userId;
+    }
+
+    public String username() {
+        return this.username;
+    }
+
+    public static TokenPayload fromMap(Map<String, Object> map) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/auth/TokenPayload.java:40]: delete this method

**Review by javeme — APPROVED** [review]

**Review by javeme — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Pull request merged**